cf_units/cf_units.py

```diff
@@ -1871,8 +1914,9 @@ def convert(self, value, other, ctype=FLOAT64):
                         if issubclass(value_copy.dtype.type, np.integer):
                             value_copy = value_copy.astype(
                                 _ctypes2numpy[ctype])
-                        # strict type check of numpy array
-                        if value_copy.dtype.type not in _numpy2ctypes.keys():
+                        # Strict type check of numpy array.
+                        keys = _numpy2ctypes.keys()
```

Comment:
This can be simplified further:
```python

                         if value_copy.dtype.type not in _numpy2ctypes:
                             raise TypeError(
                                 "Expect a numpy array of '%s' or '%s'" %
                                 tuple(sorted(_numpy2ctypes.keys())))
```
To check if a value is a key in a dictionary, you can just do:
```python
d = {'a': 1, 'b': 2}
'a' in d  # <-- True
# is equivalent to
'a' in d.keys()

'aa' not in d  # <-- True
# is equivalent to
'aa' not in d.keys()
```

same in line 1932.